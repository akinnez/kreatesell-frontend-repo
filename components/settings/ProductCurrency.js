import React,{useState,useEffect} from 'react'
import style from './Index.module.scss'
import {Checkbox,Row,Col,Spin} from 'antd'
import {Button} from '../form-input'
import ApiService from '../../utils/axios'



const Index = ({list=[]})=>{
  

    return(
        <div className={style.wrapper}>

        <h3>Store Currency Settings</h3>
        <div className={style.bordered}>
            <h4>Custom Product Currency - Customize your product currency</h4>
            <p>Your(Creators) country currency is selected by default but you can decide to turn it off if you prefer to. You can select other options if you prefer to set the amount when adding a product on the add product section. Anyone you do not select here will be converted to automatically if allowed and selected by your customer.</p>
            <h4>Customize the amount you can set when adding a product</h4>

            <Checkbox.Group style={{ width: '100%' }} onChange={(e)=>console.log(e)}>
                <Row>
                      {  list?.map(({label,value},i)=>(
                            <Col key={i}  md={8} sm={8}>
                            <Checkbox value={value}>{label}</Checkbox>
                    </Col>
                        ))}
                    
                </Row>
            </Checkbox.Group>

            <Button type="primary" style={{marginTop:"20px"}} label="Update Details"/>
       
            </div>
        
        </div>
    )
}

export default Index