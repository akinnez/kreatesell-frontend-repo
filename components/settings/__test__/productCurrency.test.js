import renderer from "react-test-renderer";
import ProductCurrency from "../ProductCurrency";

const mockList = [
  { label: "mockLabel1", value: 1 },
  { label: "mockLabel2", value: 2 },
  { label: "mockLabel3", value: 3 },
];

describe("ProductCurrency : ", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it("renders a snapshot of <ProductCurrency /> with no props", () => {
    const productCurrency = renderer.create(<ProductCurrency />).toJSON();
    expect(productCurrency).toMatchSnapshot();
  });

  it("renders a snapshot of <ProductCurrency /> with mock list", () => {
    const productCurrency = renderer
      .create(<ProductCurrency list={mockList} />)
      .toJSON();
    expect(productCurrency).toMatchSnapshot();
  });
});
