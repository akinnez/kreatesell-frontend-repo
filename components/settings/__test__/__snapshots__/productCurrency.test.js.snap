// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCurrency :  renders a snapshot of <ProductCurrency /> with mock list 1`] = `
<div
  className="wrapper"
>
  <h3>
    Store Currency Settings
  </h3>
  <div
    className="bordered"
  >
    <h4>
      Custom Product Currency - Customize your product currency
    </h4>
    <p>
      Your(Creators) country currency is selected by default but you can decide to turn it off if you prefer to. You can select other options if you prefer to set the amount when adding a product on the add product section. Anyone you do not select here will be converted to automatically if allowed and selected by your customer.
    </p>
    <h4>
      Customize the amount you can set when adding a product
    </h4>
    <div
      className="ant-checkbox-group"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-sm-8 ant-col-md-8"
          style={Object {}}
        >
          <label
            className="ant-checkbox-wrapper"
          >
            <span
              className="ant-checkbox"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-checkbox-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="checkbox"
                value={1}
              />
              <span
                className="ant-checkbox-inner"
              />
            </span>
            <span>
              mockLabel1
            </span>
          </label>
        </div>
        <div
          className="ant-col ant-col-sm-8 ant-col-md-8"
          style={Object {}}
        >
          <label
            className="ant-checkbox-wrapper"
          >
            <span
              className="ant-checkbox"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-checkbox-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="checkbox"
                value={2}
              />
              <span
                className="ant-checkbox-inner"
              />
            </span>
            <span>
              mockLabel2
            </span>
          </label>
        </div>
        <div
          className="ant-col ant-col-sm-8 ant-col-md-8"
          style={Object {}}
        >
          <label
            className="ant-checkbox-wrapper"
          >
            <span
              className="ant-checkbox"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-checkbox-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="checkbox"
                value={3}
              />
              <span
                className="ant-checkbox-inner"
              />
            </span>
            <span>
              mockLabel3
            </span>
          </label>
        </div>
      </div>
    </div>
    <button
      className="ant-btn ant-btn-primary btn"
      onClick={[Function]}
      style={
        Object {
          "marginTop": "20px",
        }
      }
      type="button"
    >
      <span>
        Update Details
      </span>
    </button>
  </div>
</div>
`;

exports[`ProductCurrency :  renders a snapshot of <ProductCurrency /> with no props 1`] = `
<div
  className="wrapper"
>
  <h3>
    Store Currency Settings
  </h3>
  <div
    className="bordered"
  >
    <h4>
      Custom Product Currency - Customize your product currency
    </h4>
    <p>
      Your(Creators) country currency is selected by default but you can decide to turn it off if you prefer to. You can select other options if you prefer to set the amount when adding a product on the add product section. Anyone you do not select here will be converted to automatically if allowed and selected by your customer.
    </p>
    <h4>
      Customize the amount you can set when adding a product
    </h4>
    <div
      className="ant-checkbox-group"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="ant-row"
        style={Object {}}
      />
    </div>
    <button
      className="ant-btn ant-btn-primary btn"
      onClick={[Function]}
      style={
        Object {
          "marginTop": "20px",
        }
      }
      type="button"
    >
      <span>
        Update Details
      </span>
    </button>
  </div>
</div>
`;
